p2p/src/net/mock/transport/traits.rs

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2022 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::{
+    fmt::{Debug, Display},
+    hash::Hash,
+};
+
+use async_trait::async_trait;
+
+use crate::net::mock::types::Message;
+
+// TODO: FIXME: Better description.
+/// An abstraction layer for creating "sockets" ("connections").
+#[async_trait]
+pub trait TransportService: Sized {
+    type Socket: Send;
+
+    // TODO: FIXME: Remove Hash and Display?
```

Comment:
If you want to relax these trait bounds, I think you also need to relax them in the `NetworkService` that is defined in `src/net/mod.rs`. I'm indifferent about removing `Hash` if you want to start using B trees but I don't know why you'd want to remove `Display`.

Reply:
I'm not insisting on these changes, but I thought that we prefer not to use `std::hash::Hash` in order to avoid confusion between cryptographic hashes that we use everywhere. As for `Display`, I think this bound is too restrictive, so `Debug` should be enough.